### PR TITLE
ENT-2792: Port ConsumerContentOverrideResource to spec-first

### PR DESCRIPTION
**Pull request by Januson**

### CODE REVIEW

File: api/candlepin-api-spec.yaml [hunk 1]
```diff
@@ -509,6 +509,166 @@ paths:
...
+      responses:
+        200:
+          description: List of content overrides of the Consumer
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ContentOverrideDTO'
+        400:
```
Comment (wolfdale) [inline api/candlepin-api-spec.yaml:539]: One minor change here as I think it should be ```404``` originating from ```verifyAndLookupConsumer``` method of consumer curator class.

**Review by wolfdale — CHANGES_REQUESTED** [review]
All changes looks good to me.

File: api/candlepin-api-spec.yaml (hunk 1 above)
Comment (Januson) [reply api/candlepin-api-spec.yaml]: Done.

**Review by wolfdale — APPROVED** [review]
ACK

**Pull request merged**